landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+    ...     mg,
+    ...     surface="topographic__elevation",
+    ...     flow_director="FlowDirectorD8",
+    ...     runoff_rate=None,
+    ...     depression_finder=None,
+    ... )
+    >>> latero = LateralEroder(mg, latero_mech="UC", Kv=0.001, Kl_ratio=1.5)
+
+    Run one step of flow accumulation and lateral erosion to get the dzlat array
+    needed for the next part of the test.
+
+    >>> fa.run_one_step()
+    >>> mg, dzlat = latero.run_one_step(dt)
+
+    Evolve the landscape until the first occurence of lateral erosion. Save arrays
+    volume of lateral erosion and topographic elevation before and after the first
+    occurence of lateral erosion
+
+    >>> while min(dzlat) == 0.0:
+    ...     oldlatvol = mg.at_node["volume__lateral_erosion"].copy()
+    ...     oldelev = mg.at_node["topographic__elevation"].copy()
+    ...     fa.run_one_step()
+    ...     mg, dzlat = latero.run_one_step(dt)
+    ...     newlatvol = mg.at_node["volume__lateral_erosion"]
+    ...     newelev = mg.at_node["topographic__elevation"]
+    ...     mg.at_node["topographic__elevation"][mg.core_nodes] += U * dt
+
+    Before lateral erosion occurs, *volume__lateral_erosion* has values at
+    nodes 6 and 10.
+
+    >>> np.around(oldlatvol, decimals=0)
+    array([  0.,   0.,   0.,   0.,
+             0.,   0.,  79.,   0.,
+             0.,   0.,  24.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.])
+
+
+    After lateral erosion occurs at node 6, *volume__lateral_erosion* is reset to 0
+
+    >>> np.around(newlatvol, decimals=0)
+    array([  0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,  24.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.])
+
+
+    After lateral erosion at node 6, elevation at node 6 is reduced by -1.41
+    (the height stored in dzlat[6]).
+
+    >>> np.around(oldelev, decimals=2)
+    array([ 0.  ,  1.03,  2.03,  3.  ,
+            1.04,  1.77,  2.45,  4.08,
+            2.09,  2.65,  3.18,  5.09,
+            3.04,  3.65,  4.07,  6.03,
+            4.03,  5.09,  6.02,  7.1 ])
+
+    >>> np.around(newelev, decimals=2)
+    array([ 0.  ,  1.03,  2.03,  3.  ,
+            1.04,  1.77,  1.03,  4.08,
+            2.09,  2.65,  3.18,  5.09,
+            3.04,  3.65,  4.07,  6.03,
+            4.03,  5.09,  6.02,  7.1 ])
+
+    >>> np.around(dzlat, decimals=2)
+    array([ 0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  , -1.41,  0.  ,
+            0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  ,  0.  ,  0. ])
+    """
+
+    _name = "LateralEroder"
+
+    _input_var_names = (
+        "topographic__elevation",
+        "drainage_area",
+        "flow__receiver_node",
+        "flow__upstream_node_order",
+        "topographic__steepest_slope",
+    )
+
+    _output_var_names = ("topographic__elevation",
+         "dzlat",
+         "vollat",
+         "qs_in"
+     )
```

Comment:
@gregtucker @mcflugen I think we need to provide some more guidance regarding what are appropriate field names. For example I don't think `dzlat` is appropriate, but we haven't really said that anywhere in the developer docs. 

We should assist in providing guidance about such names. 

Also tagging @SiccarPoint.

Reply:
@gregtucker and I were just discussing some possible names to propose. 

[this is the page in the docs](https://github.com/landlab/landlab/wiki/Components#landlab-standard-naming-conventions) where we talk a bit about naming conventions. 

I find standard names really hard to implement in practice, so I'd be very interested in @SiccarPoint and @mcflugen's comments here.

Some proposed names to start from:

dzlat: lateral_erosion__topographic_elevation__change or
lateral_erosion__topographic_elevation__rate_of_change
# note the until listed for this are m/y, but your variable description implies dimensions of ?

vollat: lateral_erosion__volume_eroded
qs_in: sediment__flux

Reply:
I did not change any of the variable names, pending comments/direction from others, but I did change the description of dzlat to reflect this variable is in units of length.

Reply:
I think the above are fine choices for alternative names. For dzlat, I think even something slightly shorter like "lateral_erosion__depth_increment" would be fine as well. We don't have strict guidelines or pure standardization regarding names. The general qualitative guide line is to have it be somewhat descriptive, but not too long (where the "somewhat" and "too" are fuzzy).